code/modules/hallucination/blood_flow.dm

```diff
@@ -0,0 +1,69 @@
+/datum/hallucination/blood_flow
+	random_hallucination_weight = 3
+	hallucination_tier = HALLUCINATION_TIER_COMMON
+	/// The bleeding hallucination's image
+	var/image/bleeding
+
+/datum/hallucination/blood_flow/start()
+	if(!hallucinator.client || !iscarbon(hallucinator))
+		return FALSE
+
+	var/mob/living/carbon/carb_hallucinator = hallucinator
+	if(!length(carb_hallucinator.bodyparts) || HAS_TRAIT(carb_hallucinator, TRAIT_NOBLOOD))
```

Comment:
I looked at this, wondered if returning false rerolls the hallucination, and found that this comment you left 2 years ago still remains. Sad!
```
	// It's not guaranteed that the hallucination passed can successfully be initiated.
	// This means there may be cases where someone should have a hallucination but nothing happens,
	// notably if you pass a randomly picked hallucination type into this.
	// Maybe there should be a separate proc to reroll on failure?
```

Won't require it for this PR though.

Reply:
😉